service/matching/matchingEngine_test.go

```diff
@@ -1587,6 +1588,10 @@ func (s *matchingEngineSuite) TestTaskQueueManagerGetTaskBatch_ReadBatchDone() {
 
 	tlMgr.Start()
 
+	// tlMgr.taskWriter startup is async so give it time to complete, otherwise
+	// the following few lines get clobbered as part of the taskWriter.Start()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
For the record I hate this. But with the way the existing tests reach into the private implementation of the code I don't see another way.

Reply:
Just want to make sure this would not cause flaky test.

Reply:
It's not great and it might make for spurious failures. If you can think of another way to do the same thing I'm all ears. But apart from rewriting the test and restructuring a lot of the code, I don't see another way.